testengine/manglers.go

```diff
@@ -29,11 +43,34 @@ func (sm *SilencingMangler) BeforeStep(random int, el *EventLog) {
 		return
 	}
 
-	if el.NextEventLogEntry.Prev != nil {
-		el.NextEventLogEntry.Prev.Next = currentEntry.Next
+	DropCurrent(el)
+}
+
+// DroppyMangler will randomly delete event which is a 'Receive' from
+// a particular droppy node.  It is effectively as if all packets
+// are being randomly dropped from this source. The drop percentage is
+// decided by DropRate (1-100)
+type DroppyMangler struct {
+	NodeToDrop uint64
+	DropRate   int // percentage of events to drop, 1-100
+}
+
+func (dm *DroppyMangler) BeforeStep(random int, el *EventLog) {
+	currentEntry := el.NextEventLogEntry
+
+	recv, ok := currentEntry.Event.Type.(*tpb.Event_Receive_)
+	if !ok {
+		return
 	}
-	if currentEntry.Next != nil {
-		currentEntry.Next.Prev = currentEntry.Prev
+
+	if recv.Receive.Source != dm.NodeToDrop {
+		return
 	}
-	el.NextEventLogEntry = currentEntry.Next
+
+	x := math.MaxInt64 / 100
```

Comment:
Since the random type is an int, and not an int64, we're not really guaranteed to know the magnitude of it.  We could change the code (it would be trivial) to exactly specify the size of the rand to generate, but, for just this case I'd proopose skipping the math.Max stuff and doing:

```
if random % 100 > dm.DropRate
```

Seems to accomplish the same thing more clearly with less code?